client/components/auth/ResetPassword.vue

```diff
@@ -1,77 +1,93 @@
-      color="pink lighten-1"
-      text
-      class="mb-5">
+    <v-alert :value="!!error" color="pink lighten-1" text class="mb-5">
       {{ error }}
     </v-alert>
+    <v-progress-circular v-if="isLoading" color="primary darken-2" indeterminate />
     <validation-observer
+      v-else
       ref="form"
       @submit.prevent="$refs.form.handleSubmit(submit)"
       tag="form"
       novalidate>
-      <validation-provider
-        v-slot="{ errors }"
-        vid="password"
-        name="password"
-        rules="required|alphanumerical|min:6">
-        <v-text-field
-          v-model="password"
-          :error-messages="errors"
-          type="password"
+      <template v-if="!error">
+        <validation-provider
+          v-slot="{ errors }"
+          vid="password"
           name="password"
-          label="Password"
-          placeholder="Password"
-          prepend-inner-icon="mdi-lock"
-          outlined
-          class="required mb-1" />
-      </validation-provider>
-      <validation-provider
-        v-slot="{ errors }"
-        vid="passwordConfirmation"
-        name="password"
-        rules="required|confirmed:password">
-        <v-text-field
-          v-model="passwordConfirmation"
-          :error-messages="errors"
-          type="password"
-          name="passwordConfirmation"
-          label="Re-enter password"
-          placeholder="Password confirmation"
-          prepend-inner-icon="mdi-lock-outline"
-          outlined
-          class="required" />
-      </validation-provider>
-      <v-btn
-        type="submit"
-        color="primary darken-4"
-        rounded block dark
-        class="my-1">
-        Change password
-      </v-btn>
+          rules="required|alphanumerical|min:6">
+          <v-text-field
+            v-model="password"
+            :error-messages="errors"
+            type="password"
+            name="password"
+            label="Password"
+            placeholder="Password"
+            prepend-inner-icon="mdi-lock"
+            outlined
+            class="required mb-1" />
+        </validation-provider>
+        <validation-provider
+          v-slot="{ errors }"
+          vid="passwordConfirmation"
+          name="password"
+          rules="required|confirmed:password">
+          <v-text-field
+            v-model="passwordConfirmation"
+            :error-messages="errors"
+            type="password"
+            name="passwordConfirmation"
+            label="Re-enter password"
+            placeholder="Password confirmation"
+            prepend-inner-icon="mdi-lock-outline"
+            outlined
+            class="required" />
+        </validation-provider>
+        <v-btn
+          type="submit"
+          color="primary darken-4"
+          rounded block dark
+          class="my-1">
+          Change password
+        </v-btn>
+      </template>
+      <router-link v-else :to="{ name: 'forgot-password' }" class="forgot-password">
```

Comment:
Consider rewriting this in a way you are not toggling part of the form (this should be outside).